mainnet_contract/contracts/Subnet.sol

```diff
@@ -0,0 +1,214 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.4.21 <0.9.0;
+pragma experimental ABIEncoderV2;
+
+import "./HeaderReader.sol";
+
+contract Subnet {
+
+  struct Header {
+    bytes32 hash;
+    int number;
+    uint64 round_num;
+    bytes32 parent_hash;
+    bool finalized;
+    uint mainnet_num;
+    bytes src;
+  }
+
+  struct Validators {
+    address[] set;
+    int threshold;
+  }
```

Comment:
two separate setting method. address[] is from new Header.Validators.NextEpoch. threshold is from (grand)master, so there should be a new func SetThreshold(at int, threhold int)